### PR TITLE
Auth  functionality 

### PR DESCRIPTION
**Pull request by stutisk**
added userpage
logout page 
and auth functionality 
added empty state screen

### CODE REVIEW

File: src/Pages/Authpages/Login.jsx [hunk 1]
```diff
@@ -1,74 +1,122 @@
...
+        Email: response.data.foundUser.email,
+        name:
+          response.data.foundUser.firstName +
+          " " +
+          response.data.foundUser.lastName,
+      };
+      localStorage.setItem("token", response.data.encodedToken);
+      localStorage.setItem("user", JSON.stringify(userDetail));
+    } catch (error) {
+      console.log(error)
```
Comment (Varun-28) [inline src/Pages/Authpages/Login.jsx:36]: - Working as intended to work
- ok to have console.logs now, but must remove them at the time of final production
- you can use built-in react-alert npm packages for showing alert messages at the time like -  error, success login etc (suggestion you can do it in later branch)
- code is clean and easy to understand

File: src/App.js [hunk 2]
```diff
@@ -2,7 +2,20 @@ import MockAPI from "./components/Mockman/Mockman";
...
+  Homepage,
+  Likedvideos,
+  Playlist,
+  Videolistingpage,
+  Watchlater,
+  Login,
+  Signup,
+  Userpage,
+  Logout
+} from "./Pages/index";
```
Comment (abhij1607) [inline src/App.js:16]: No need to mention index in barrel imports

Comment (stutisk) [reply src/App.js:16]: didn't know about this , thanku

File: src/Pages/Authpages/Login.jsx (hunk 1 above)
Comment (stutisk) [reply src/Pages/Authpages/Login.jsx:36]: yes will be implementing alerts ,thnku

**Review by abhij1607 — APPROVED** [review]
Neat and clean code
Nice to see use of useRef
Can be merged

Suggestions

- cleanup console.log
- Nested route could be used for RequiresAuth

**Pull request merged**